### PR TITLE
Support for override folder is added

### PR DESCRIPTION
**Pull request by critto-bg**
This is a bit bulky, but it's a single piece of work in my opinion (the stream object + core functions to interact with the folder and support inside the key built-in functions).

### CODE REVIEW

File: infinity/core.py [hunk 1]
```diff
@@ -508,6 +512,43 @@ def find_file (filename, type = None):
 
     return None
 
+def locate_override ():
+    dir = os.path.join (game_dir, "override")
+
+    if (os.path.exists (dir)):
```
Comment (lynxlynxlynx) [inline infinity/core.py]: isn't this case sensitive? We should look for at least Override too.

File: infinity/stream.py [hunk 2]
```diff
@@ -510,6 +511,43 @@ def load_object (self):
...
+        if len (obj) > 1 and self.type is None:
+            raise RuntimeError, name + ": more than one result in override, types " + ' ' . join([ str(o['type']) for o in obj ])
+
+        if len (obj) == 0:
+            raise RuntimeError, name + ": not found in override!"
+
+        try:
+            obj = obj[index]
+        except IndexError:
+            obj = obj[0]
```
Comment (lynxlynxlynx) [inline infinity/stream.py]: I can't imagine where passing the index would be needed, but still, silently using a fallback doesn't feel nice.

**Comment by critto-bg** [comment]
Fixed the issues, please, take a look.

**Comment by lynxlynxlynx** [comment]
Looks good to me. I think "OVERRIDE" should never happen. @edheldil  mentioned core.find_file(), but I don't see any potential for it here.

**Comment by critto-bg** [comment]
So, shall we merge it or do you think anything else is needed?

**Comment by lynxlynxlynx** [comment]
Thanks. :+1:

**Pull request merged**